src/main/java/org/opensearch/security/identity/SecurityTokenManager.java

```diff
@@ -131,6 +141,33 @@ public ExpiringBearerAuthToken issueOnBehalfOfToken(final Subject subject, final
         }
     }
 
+    public ExpiringBearerAuthToken issueApiToken(
+        final String name,
+        final Long expiration,
+        final List<String> clusterPermissions,
+        final List<ApiToken.IndexPermission> indexPermissions
+    ) {
+        final User user = threadPool.getThreadContext().getTransient(ConfigConstants.OPENDISTRO_SECURITY_USER);
+        if (user == null) {
```

Comment:
in what cases will this be nul?? super-admin or plugins?

Reply:
It should only be null if the threadcontext is stashed. Since I am always calling it with the user in the threadcontext, I removed the null check

Reply:
In: [09018dc](https://github.com/opensearch-project/security/pull/4967/commits/09018dc5dce3143acb896088887550956874175c)